pdf/pdfcanvas.cpp

```diff
@@ -267,22 +278,48 @@ void PDFCanvas::layout()
     update();
 }
 
+static qreal dist2FromRect(const QRectF &rect, const QPoint &point)
+{
+    qreal dist = 0.;
+
+    if ((qreal)point.x() < rect.x()) {
+        dist += (rect.x() - (qreal)point.x()) * (rect.x() - (qreal)point.x());
+    } else if ((qreal)point.x() > rect.x() + rect.width()) {
+        dist += (rect.x() + rect.width() - (qreal)point.x())
+                * (rect.x() + rect.width() - (qreal)point.x());
+    }
+    if ((qreal)point.y() < rect.y()) {
+        dist += (rect.y() - (qreal)point.y()) * (rect.y() - (qreal)point.y());
+    } else if ((qreal)point.y() > rect.y() + rect.height()) {
+        dist += (rect.y() + rect.height() - (qreal)point.y())
+                * (rect.y() + rect.height() - (qreal)point.y());
+    }
+
```

Comment:
I think manhattan distance would be both simpler as code and as specification for function.